datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/VersionedDatasetMapper.java

```diff
@@ -65,24 +67,24 @@ public VersionedDataset apply(@Nonnull final EntityResponse entityResponse) {
     SystemMetadata schemaSystemMetadata = getSystemMetadata(aspectMap, SCHEMA_METADATA_ASPECT_NAME);
 
     mappingHelper.mapToResult(DATASET_KEY_ASPECT_NAME, this::mapDatasetKey);
-    mappingHelper.mapToResult(DATASET_PROPERTIES_ASPECT_NAME, this::mapDatasetProperties);
+    mappingHelper.mapToResult(DATASET_PROPERTIES_ASPECT_NAME, (entity, dataMap) -> this.mapDatasetProperties(entity, dataMap, entityUrn));
     mappingHelper.mapToResult(DATASET_DEPRECATION_ASPECT_NAME, (dataset, dataMap) ->
         dataset.setDeprecation(DatasetDeprecationMapper.map(new DatasetDeprecation(dataMap))));
     mappingHelper.mapToResult(SCHEMA_METADATA_ASPECT_NAME, (dataset, dataMap) ->
-        dataset.setSchema(SchemaMapper.map(new SchemaMetadata(dataMap), schemaSystemMetadata)));
+        dataset.setSchema(SchemaMapper.map(new SchemaMetadata(dataMap), schemaSystemMetadata, entityUrn)));
     mappingHelper.mapToResult(EDITABLE_DATASET_PROPERTIES_ASPECT_NAME, this::mapEditableDatasetProperties);
     mappingHelper.mapToResult(VIEW_PROPERTIES_ASPECT_NAME, this::mapViewProperties);
     mappingHelper.mapToResult(INSTITUTIONAL_MEMORY_ASPECT_NAME, (dataset, dataMap) ->
         dataset.setInstitutionalMemory(InstitutionalMemoryMapper.map(new InstitutionalMemory(dataMap))));
     mappingHelper.mapToResult(OWNERSHIP_ASPECT_NAME, (dataset, dataMap) ->
-        dataset.setOwnership(OwnershipMapper.map(new Ownership(dataMap))));
+        dataset.setOwnership(OwnershipMapper.map(new Ownership(dataMap), entityUrn)));
     mappingHelper.mapToResult(STATUS_ASPECT_NAME, (dataset, dataMap) ->
         dataset.setStatus(StatusMapper.map(new Status(dataMap))));
-    mappingHelper.mapToResult(GLOBAL_TAGS_ASPECT_NAME, this::mapGlobalTags);
+    mappingHelper.mapToResult(GLOBAL_TAGS_ASPECT_NAME, (dataset, dataMap) -> this.mapGlobalTags(dataset, dataMap, entityUrn));
```

Comment:
I was considering this, but I think adding the `entityUrn` to the pdl is too heavy a change. Most significantly, it would mean that the generated python classes datahub users reference from the emitter apis would request the `entityUrn` to be populated, which I imagine would be very confusing.

Reply:
Not sure I'm following the part about Python emitters. But instead of hardcoding the URN context in why not just add a MapperContext object that we can continue to extend over time should we need to?

This feels too hardcoded to URNs. 

Imagine that in the future we need the current user context - now what? Do we add a new argument to every mapper again? I mean...